[FIX] website_livechat: public user lead creation rights

nd-dew opened the pull request:
ChatBot can't create lead, because it can't access website.visitor, as a public user. In order to allow lead creation from given email and phone this commit adds sudo rights for this operation.

[Reproduce]
- Install: website_crm_livechat
- Go to: Live Chat / Channels -> "Configure a channel"
	- Modify Channel Rules for "/contactus" regex to use Lead Generation Bot
- Go to Live Chat / Configuration / Chatbots  -> Lead Generation Bot script change to:
	1. Message: "e", Step Type "Email"
	2. Message: "p", Step Type "Phone"
	3. Message: "c", Step Type "Create lead"
- log out (or use incognito) -> go to contact us page
- open chat, answer its questions (email, phone)
- BUG: Access Error msg

opw-3858298

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/162979).

nd-dew commented:
rebased

nd-dew commented:
## Fixing tests that were passing by sudoing
- I removed sudo from message_post to test if member is added correctly (members can post)
- Added sudo to `_process_answer` call to mimic behavior from `LivechatChatbotScriptController.chatbot_trigger_step

## That however breaks test for a portal user
So I guess now we're onto that :face_exhaling:

nd-dew commented:
Team adopted different solution
https://github.com/odoo/odoo/pull/169975
